run_integration_tests.go

```diff
@@ -0,0 +1,189 @@
+// +build ignore
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+)
+
+type CIEnvironment interface {
+	Prepare()
+	RunTests()
+}
+
+type TravisEnvironment struct {
+	goxArch []string
+	goxOS   []string
+}
+
+func (env *TravisEnvironment) Prepare() {
+	msg("preparing environment for Travis CI\n")
```

Comment:
I don't see anything specific to Travis in either this function nor the other one.. How about we do a `s/TravisEnvironment/UnixTestEnvironment/g` on this whole file?